### DISCUSSION TITLE
createTrigger support for different dialects?

### DISCUSSION THREAD

**Discussion by BananaMasterz** [simolus3/drift#2593] [General]
I tried using `createTrigger` using `mariadb` dialect and I got 
```
'package:drift/src/runtime/query_builder/helpers.dart': Failed assertion: line 28 pos 7: 'sql.containsKey(dialect)': Tried running SQL optimized for the following dialects: Closure: ([String]) => String from Function 'join':.. However, the database is running SqlDialect.mariadb. Has that dialect been added to the `dialects` drift builder option?
```
(idk how I forgot to check trigger creation in my pull request on adding mariadb support lol)
So now I'm not sure what I should do. I think the syntax is the same in sqlite and mariadb so it should be something simple? Something to do with `createStatementsByDialect`? not sure...

(I do have `mariadb` in my build.yaml in case that's not obvious)

**Comment by simolus3** [comment]
Can you show the generated code for the trigger field? Does it use `Trigger` or `Trigger.byDialect`?

**Reply by BananaMasterz** [reply]
Ah damn, I'm stupid... didn't realize `Trigger.byDialect` exists. Works that way. Thanks!
